### PR TITLE
Sk0 102

### PR DESCRIPTION
**Pull request by markkovari**
Should be merged after #86

### CODE REVIEW

**Review by bendesbeus — APPROVED** [review]

**Review by rolcho — APPROVED** [review]

**Review by rolcho — APPROVED** [review]

File: e2e/features/steps/login/common.steps.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+Given('the user clicks on the {string} button', async function (text: string) {
+    //There should be a better way to do this, but there might be an issue with the mat-button selector
+    //Should be the same as in the ../register/common.steps.ts
+    const buttonSelector = `#main-content > ion-router-outlet > app-login > ion-content > ion-grid > ion-row > ion-col > ion-card > ion-card-content > ion-button`
+    const selector = await page.waitForSelector(buttonSelector);
+    await selector.click();
+});
+
+Then('the page should navigate to the register page', async function () {
+    expect(page.url()).toBe(`${baseUrl}/register`)
```
Comment (Thomas851204) [inline e2e/features/steps/login/common.steps.ts]: For me the e2e test throws the following error:
```
1) Scenario: User navigates to register # features/login.feature:3
   ✔ Given the user is on the login page # features/steps/login/common.steps.ts:6
   ✔ And the user clicks on the "Még nincs fiókja?" button # features/steps/login/common.steps.ts:10
   ✖ Then the page should navigate to the register page # features/steps/login/common.steps.ts:18
       Error: expect(received).toBe(expected) // Object.is equality
       
       Expected: "http://localhost:8080/register"
       Received: "http://localhost:8080/login"
           at Proxy.<anonymous> (/workspaces/sk0/e2e/node_modules/playwright/lib/matchers/expect.js:174:37)
           at World.<anonymous> (/workspaces/sk0/e2e/features/steps/login/common.steps.ts:19:24)
           at /workspaces/sk0/e2e/features/steps/login/common.steps.ts:8:71
           at __awaiter (/workspaces/sk0/e2e/features/steps/login/common.steps.ts:4:12)
           at World.<anonymous> (/workspaces/sk0/e2e/features/steps/login/common.steps.ts:30:12)
```

If I use ```await page.waitForUrl(`${baseUrl}/register`)``` then it works.

Before running the test I follow the docker compose up + npm i (in e2e) method.

File: e2e/features/steps/register/common.steps.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      .fill(email);
+  },
+);
+
+When(
+  "the user clicks the {string} button",
+  async function (buttonLabel: string) {
+    const submitButtonLocator = page
+      .getByRole("button", { name: buttonLabel })
+      .first();
```
Comment (Thomas851204) [inline e2e/features/steps/register/common.steps.ts]: Im getting the followin error:
```
2) Scenario: Cancel registration # features/registration.feature:22
   ✔ Given the user is on the registration page # features/steps/register/common.steps.ts:34
   ✔ When the user enters a name: "validName" # features/steps/register/common.steps.ts:6
   ✔ And the user enters an email address: "valid@email.com" # features/steps/register/common.steps.ts:14
   ✔ And the user enters a password: "ValidPassword123" # features/steps/register/common.steps.ts:10
   ✖ And the user clicks the "JELENTKEZZEN BE!" button # features/steps/register/common.steps.ts:24
       Error: function timed out, ensure the promise resolves within 5000 milliseconds
           at Timeout.<anonymous> (/workspaces/sk0/e2e/node_modules/@cucumber/cucumber/src/time.ts:52:14)
           at listOnTimeout (node:internal/timers:573:17)
           at processTimers (node:internal/timers:514:7)
   - Then the page should be naviagted to the login page # features/steps/register/cancel.steps.ts:5
 ```
I cant resolve this. No matter what selector or locator I use with any kind of parameters (even tried giving data-testid for both buttons), it wont work for both "Regisztráció" and "Jelentkezzen be" buttons.

**Review by Thomas851204 — CHANGES_REQUESTED** [review]
One issue with awaiting the the page to load and one issue with button selector in registration's common steps.


Just my 2 cents: I understand the utility of resuing steps in separate common.steps files, however for me the downside is that it makes reading the actual test code harder, though I guess that is subjective and one can get used to it.

File: e2e/features/steps/register/common.steps.ts (hunk 2 above)
Comment (rolcho) [reply e2e/features/steps/register/common.steps.ts]: I made the universal [Button click](https://github.com/cimem-hu/sk0/pull/93/files) function in the common.steps.ts and it absolutely doesn't matter if it is Given or Then.

File: e2e/features/login.feature [hunk 3]
```diff
@@ -0,0 +1,6 @@
+Feature: Login
+
+  Scenario: User navigates to register
+    Given the user is on the login page
+    And the user clicks on the "Még nincs fiókja?" button
```
Comment (rolcho) [inline e2e/features/login.feature]: The text is: "Nincs még fiókja?"
```suggestion
    And the user clicks on the "Nincs még fiókja?" button
```

Comment (markkovari) [reply e2e/features/login.feature]: Thanks, would not understand what is going on.

File: e2e/features/steps/register/common.steps.ts (hunk 2 above)
Comment (markkovari) [reply e2e/features/steps/register/common.steps.ts]: @Thomas851204 you are correct, let me create a commit where I am rearranging the steps, it is a bit counter intuitive now, needs some love

ps.: @rolcho already stole your way ;) of selecting the buttons

File: e2e/features/steps/login/common.steps.ts (hunk 1 above)
Comment (markkovari) [reply e2e/features/steps/login/common.steps.ts]: Worked for me without it as well, but added to both place where we are checking the stat of the url.

Comment (markkovari) [reply e2e/features/steps/login/common.steps.ts]: Thats the other solution
``` await page.waitForLoadState("networkidle")```

File: e2e/features/steps/common.steps.ts [hunk 4]
```diff
@@ -21,13 +21,12 @@ When(
   },
 );
 
-When(
-  "the user clicks the {string} button",
+Then(
+  `the user clicks the {string} button`,
```
Comment (markkovari) [inline e2e/features/steps/common.steps.ts]:
```suggestion
When(
  "the user clicks the {string} button",
```

**Review by markkovari — COMMENTED** [review]
👯

**Review by rolcho — APPROVED** [review]
🎢

**Review by Thomas851204 — APPROVED** [review]
New changes are pretty straight forward, good job on unification and the yml update!

Good to go.

**Review by Thomas851204 — APPROVED** [review]

**Pull request merged**